src/Qowaiv.Validation.Abstractions/ResultActExtensions.cs

```diff
@@ -129,16 +104,146 @@ public static async Task<Result<TModel>> ActAsync<TModel>(this Task<Result<TMode
             _ = Guard.NotNull(promise, nameof(promise));
             Guard.NotNull(action, nameof(action));
 
-            var result = await promise.ConfigureAwait(false);
+            var result = await promise.ContinueOnAnyContext();
+            return result is null
+                ? Result.For<TModel>(default)
+                : await result.ActAsync(action).ContinueOnAnyContext();
+        }
+
+        /// <summary>Invokes the action when <see cref="Result{TModel}"/> is valid.</summary>
+        /// <param name="promise">
+        /// The promise of the result.
+        /// </param>
+        /// <param name="update">
+        /// The update to apply on a successfully invoked action.
```

Comment:
The order of update/action is different compared the their order of the parameters in the function definition. Same in the two function below.